service/src/main/kotlin/org/opensearch/simpleschema/index/SimpleSearchIndex.kt

```diff
@@ -111,6 +111,40 @@ internal object SimpleSearchIndex {
         }
     }
 
+    private fun createIndexRequest(): CreateIndexRequest? {
+        val classLoader = SimpleSearchIndex::class.java.classLoader
+        val indexMappingSource = classLoader.getResource(SIMPLESCHEMA_MAPPING_FILE_NAME)?.readText()!!
+        val indexSettingsSource = classLoader.getResource(SIMPLESCHEMA_SETTINGS_FILE_NAME)?.readText()!!
+        val request = CreateIndexRequest(INDEX_NAME)
+            .mapping(indexMappingSource, XContentType.YAML)
+            .settings(indexSettingsSource, XContentType.YAML)
+        return request
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun indexCreateCall() {
+        val request = createIndexRequest()
+        try {
+            client.execute(CreateIndexAction.INSTANCE, request, object : ActionListener<CreateIndexResponse> {
+                override fun onResponse(response: CreateIndexResponse) {
+                    if (response.isAcknowledged) {
+                        log.info("$LOG_PREFIX:Index $INDEX_NAME creation Acknowledged")
+                    } else {
+                        log.warn("$LOG_PREFIX:Index $INDEX_NAME creation not Acknowledged")
+                    }
+                }
+
+                override fun onFailure(err: java.lang.Exception?) {
+                    log.error("$LOG_PREFIX:Index $INDEX_NAME creation not Acknowledged", err)
+                }
+            })
+        } catch (exception: Exception) {
+            if (exception !is ResourceAlreadyExistsException && exception.cause !is ResourceAlreadyExistsException) {
```

Comment:
Why catch a generic exception if we only care about one type?

Reply:
since the inner cause can also be ResourceAlreadyExistsException - we need to make sure both the outer & inner exceptions are not ResourceAlreadyExistsException...